Build failure: wl-screenrec

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `wl-screenrec`

### Build log

```
error: builder for '/nix/store/rqda27j6ycsb5zca3vvs4c5s5c0yg800-wl-screenrec-0.1.4-unstable-2024-07-28.drv' failed with exit code 101;
       last 25 log lines:
       > 1339~             AV_CODEC_ID_LEAD => Id::LEAD,
       > 1340~             sys::AVCodecID::AV_CODEC_ID_LC3 | sys::AVCodecID::AV_CODEC_ID_LCEVC => todo!(),
       >     |
       >
       > error[E0004]: non-exhaustive patterns: `sys::AVPacketSideDataType::AV_PKT_DATA_FRAME_CROPPING` and `sys::AVPacketSideDataType::AV_PKT_DATA_LCEVC` not covered
       >   --> /build/wl-screenrec-0.1.4-unstable-2024-07-28-vendor.tar.gz/ffmpeg-next/src/codec/packet/side_data.rs:70:15
       >    |
       > 70 |         match value {
       >    |               ^^^^^ patterns `sys::AVPacketSideDataType::AV_PKT_DATA_FRAME_CROPPING` and `sys::AVPacketSideDataType::AV_PKT_DATA_LCEVC` not covered
       >    |
       > note: `sys::AVPacketSideDataType` defined here
       >   --> /build/source/target/x86_64-unknown-linux-gnu/release/build/ffmpeg-sys-next-4f9a9cbd34c5c7ed/out/bindings.rs:3:275528
       >    |
       > 3  | ...)] pub enum AVPacketSideDataType { # [doc = " An AV_PKT_DATA_PALETTE side data packet contains exactly AVPALETTE_SIZE\n bytes worth of palette. This side data signals that a new palette is\n present."] AV_PKT_DATA_PALETTE = 0 , # [doc = " The AV_PKT_DATA_NEW_EXTRADATA is used to notify the codec or the format\n that the extradata buffer was changed and the receiving side should\n act upon it appropriately. The new extradata is embedded in the side\n data buffer and should be immediately used for processing the current\n frame or packet."] AV_PKT_DATA_NEW_EXTRADATA = 1 , # [doc = " An AV_PKT_DATA_PARAM_CHANGE side data packet is laid out as follows:\n @code\n u32le param_flags\n if (param_flags & AV_SIDE_DATA_PARAM_CHANGE_SAMPLE_RATE)\n     s32le sample_rate\n if (param_flags & AV_SIDE_DATA_PARAM_CHANGE_DIMENSIONS)\n     s32le width\n     s32le height\n @endcode"] AV_PKT_DATA_PARAM_CHANGE = 2 , # [doc = " An AV_PKT_DATA_H263_MB_INFO side data packet contains a number of\n structures with info about macroblocks relevant to splitting the\n packet into smaller packets on macroblock edges (e.g. as for RFC 2190).\n That is, it does not necessarily contain info about all macroblocks,\n as long as the distance between macroblocks in the info is smaller\n than the target payload size.\n Each MB info structure is 12 bytes, and is laid out as follows:\n @code\n u32le bit offset from the start of the packet\n u8    current quantizer at the start of the macroblock\n u8    GOB number\n u16le macroblock address within the GOB\n u8    horizontal MV predictor\n u8    vertical MV predictor\n u8    horizontal MV predictor for block number 3\n u8    vertical MV predictor for block number 3\n @endcode"] AV_PKT_DATA_H263_MB_INFO = 3 , # [doc = " This side data should be associated with an audio stream and contains\n ReplayGain information in form of the AVReplayGain struct."] AV_PKT_DATA_REPLAYGAIN = 4 , # [doc = " This side data contains a 3x3 transformation matrix describing an affine\n transformation that needs to be applied to the decoded video frames for\n correct presentation.\n\n See libavutil/display.h for a detailed description of the data."] AV_PKT_DATA_DISPLAYMATRIX = 5 , # [doc = " This side data should be associated with a video stream and contains\n Stereoscopic 3D information in form of the AVStereo3D struct."] AV_PKT_DATA_STEREO3D = 6 , # [doc = " This side data should be associated with an audio stream and corresponds\n to enum AVAudioServiceType."] AV_PKT_DATA_AUDIO_SERVICE_TYPE = 7 , # [doc = " This side data contains quality related information from the encoder.\n @code\n u32le quality factor of the compressed frame. Allowed range is between 1 (good) and FF_LAMBDA_MAX (bad).\n u8    picture type\n u8    error count\n u16   reserved\n u64le[error count] sum of squared differences between encoder in and output\n @endcode"] AV_PKT_DATA_QUALITY_STATS = 8 , # [doc = " This side data contains an integer value representing the stream index\n of a \"fallback\" track.  A fallback track indicates an alternate\n track to use when the current track can not be decoded for some reason.\n e.g. no decoder available for codec."] AV_PKT_DATA_FALLBACK_TRACK = 9 , # [doc = " This side data corresponds to the AVCPBProperties struct."] AV_PKT_DATA_CPB_PROPERTIES = 10 , # [doc = " Recommmends skipping the specified number of samples\n @code\n u32le number of samples to skip from start of this packet\n u32le number of samples to skip from end of this packet\n u8    reason for start skip\n u8    reason for end   skip (0=padding silence, 1=convergence)\n @endcode"] AV_PKT_DATA_SKIP_SAMPLES = 11 , # [doc = " An AV_PKT_DATA_JP_DUALMONO side data packet indicates that\n the packet may contain \"dual mono\" audio specific to Japanese DTV\n and if it is true, recommends only the selected channel to be used.\n @code\n u8    selected channels (0=main/left, 1=sub/right, 2=both)\n @endcode"] AV_PKT_DATA_JP_DUALMONO = 12 , # [doc = " A list of zero terminated key/value strings. There is no end marker for\n the list, so it is required to rely on the side data size to stop."] AV_PKT_DATA_STRINGS_METADATA = 13 , # [doc = " Subtitle event position\n @code\n u32le x1\n u32le y1\n u32le x2\n u32le y2\n @endcode"] AV_PKT_DATA_SUBTITLE_POSITION = 14 , # [doc = " Data found in BlockAdditional element of matroska container. There is\n no end marker for the data, so it is required to rely on the side data\n size to recognize the end. 8 byte id (as found in BlockAddId) followed\n by data."] AV_PKT_DATA_MATROSKA_BLOCKADDITIONAL = 15 , # [doc = " The optional first identifier line of a WebVTT cue."] AV_PKT_DATA_WEBVTT_IDENTIFIER = 16 , # [doc = " The optional settings (rendering instructions) that immediately\n follow the timestamp specifier of a WebVTT cue."] AV_PKT_DATA_WEBVTT_SETTINGS = 17 , # [doc = " A list of zero terminated key/value strings. There is no end marker for\n the list, so it is required to rely on the side data size to stop. This\n side data includes updated metadata which appeared in the stream."] AV_PKT_DATA_METADATA_UPDATE = 18 , # [doc = " MPEGTS stream ID as uint8_t, this is required to pass the stream ID\n information from the demuxer to the corresponding muxer."] AV_PKT_DATA_MPEGTS_STREAM_ID = 19 , # [doc = " Mastering display metadata (based on SMPTE-2086:2014). This metadata\n should be associated with a video stream and contains data in the form\n of the AVMasteringDisplayMetadata struct."] AV_PKT_DATA_MASTERING_DISPLAY_METADATA = 20 , # [doc = " This side data should be associated with a video stream and corresponds\n to the AVSphericalMapping structure."] AV_PKT_DATA_SPHERICAL = 21 , # [doc = " Content light level (based on CTA-861.3). This metadata should be\n associated with a video stream and contains data in the form of the\n AVContentLightMetadata struct."] AV_PKT_DATA_CONTENT_LIGHT_LEVEL = 22 , # [doc = " ATSC A53 Part 4 Closed Captions. This metadata should be associated with\n a video stream. A53 CC bitstream is stored as uint8_t in AVPacketSideData.data.\n The number of bytes of CC data is AVPacketSideData.size."] AV_PKT_DATA_A53_CC = 23 , # [doc = " This side data is encryption initialization data.\n The format is not part of ABI, use av_encryption_init_info_* methods to\n access."] AV_PKT_DATA_ENCRYPTION_INIT_INFO = 24 , # [doc = " This side data contains encryption info for how to decrypt the packet.\n The format is not part of ABI, use av_encryption_info_* methods to access."] AV_PKT_DATA_ENCRYPTION_INFO = 25 , # [doc = " Active Format Description data consisting of a single byte as specified\n in ETSI TS 101 154 using AVActiveFormatDescription enum."] AV_PKT_DATA_AFD = 26 , # [doc = " Producer Reference Time data corresponding to the AVProducerReferenceTime struct,\n usually exported by some encoders (on demand through the prft flag set in the\n AVCodecContext export_side_data field)."] AV_PKT_DATA_PRFT = 27 , # [doc = " ICC profile data consisting of an opaque octet buffer following the\n format described by ISO 15076-1."] AV_PKT_DATA_ICC_PROFILE = 28 , # [doc = " DOVI configuration\n ref:\n dolby-vision-bitstreams-within-the-iso-base-media-file-format-v2.1.2, section 2.2\n dolby-vision-bitstreams-in-mpeg-2-transport-stream-multiplex-v1.2, section 3.3\n Tags are stored in struct AVDOVIDecoderConfigurationRecord."] AV_PKT_DATA_DOVI_CONF = 29 , # [doc = " Timecode which conforms to SMPTE ST 12-1:2014. The data is an array of 4 uint32_t\n where the first uint32_t describes how many (1-3) of the other timecodes are used.\n The timecode format is described in the documentation of av_timecode_get_smpte_from_framenum()\n function in libavutil/timecode.h."] AV_PKT_DATA_S12M_TIMECODE = 30 , # [doc = " HDR10+ dynamic metadata associated with a video frame. The metadata is in\n the form of the AVDynamicHDRPlus struct and contains\n information for color volume transform - application 4 of\n SMPTE 2094-40:2016 standard."] AV_PKT_DATA_DYNAMIC_HDR10_PLUS = 31 , # [doc = " IAMF Mix Gain Parameter Data associated with the audio frame. This metadata\n is in the form of the AVIAMFParamDefinition struct and contains information\n defined in sections 3.6.1 and 3.8.1 of the Immersive Audio Model and\n Formats standard."] AV_PKT_DATA_IAMF_MIX_GAIN_PARAM = 32 , # [doc = " IAMF Demixing Info Parameter Data associated with the audio frame. This\n metadata is in the form of the AVIAMFParamDefinition struct and contains\n information defined in sections 3.6.1 and 3.8.2 of the Immersive Audio Model\n and Formats standard."] AV_PKT_DATA_IAMF_DEMIXING_INFO_PARAM = 33 , # [doc = " IAMF Recon Gain Info Parameter Data associated with the audio frame. This\n metadata is in the form of the AVIAMFParamDefinition struct and contains\n information defined in sections 3.6.1 and 3.8.3 of the Immersive Audio Model\n and Formats standard."] AV_PKT_DATA_IAMF_RECON_GAIN_INFO_PARAM = 34 , # [doc = " Ambient viewing environment metadata, as defined by H.274. This metadata\n should be associated with a video stream and contains data in the form\n of the AVAmbientViewingEnvironment struct."] AV_PKT_DATA_AMBIENT_VIEWING_ENVIRONMENT = 35 , # [doc = " The number of pixels to discard from the top/bottom/left/right border of the\n decoded frame to obtain the sub-rectangle intended for presentation.\n\n @code\n u32le crop_top\n u32le crop_bottom\n u32le crop_left\n u32le crop_right\n @endcode"] AV_PKT_DATA_FRAME_CROPPING = 36 , # [doc = " Raw LCEVC payload data, as a uint8_t array, with NAL emulation\n bytes intact."] AV_PKT_DATA_LCEVC = ...
       >    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        -------------------------- not covered                                                                                        ----------------- not covered
       >    = note: the matched value is of type `sys::AVPacketSideDataType`
       > help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern, a match arm with multiple or-patterns as shown, or multiple match arms
       >    |
       > 126~             AV_PKT_DATA_AMBIENT_VIEWING_ENVIRONMENT => Type::AMBIENT_VIEWING_ENVIRONMENT,
       > 127~             sys::AVPacketSideDataType::AV_PKT_DATA_FRAME_CROPPING | sys::AVPacketSideDataType::AV_PKT_DATA_LCEVC => todo!(),
       >    |
       >
       > For more information about this error, try `rustc --explain E0004`.
       > error: could not compile `ffmpeg-next` (lib) due to 5 previous errors
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix-store -l /nix/store/rqda27j6ycsb5zca3vvs4c5s5c0yg800-wl-screenrec-0.1.4-unstable-2024-07-28.drv'.
error: 1 dependencies of derivation '/nix/store/82f5z5l7ax6wjg8j8qkh8vi1c2jrzy1q-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ny689zky6hvdjxqfy5ljkpynn8d0gq40-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jva2azjfddrl2g56mkn7j9rf506wsnav-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/84s0cixh9ba3pd7jsx078gdidwsz42vm-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/041ag48avans688dsbr3yraxs28rimnr-nixos-system-nixos-24.11.20241105.a04d33c.drv' failed to build
┏━ 1 Errors: 
 ⋮ 
┃        > 1340~             sys::AVCodecID::AV_CODEC_ID_LC3 | sys::AVCodecID::AV_CODEC_ID_LCEVC => todo!(),
┃        >     |
┃        >
┃        > error[E0004]: non-exhaustive patterns: `sys::AVPacketSideDataType::AV_PKT_DATA_FRAME_CROPPING` and `sys::AVPacketSideDataType::AV_PKT_DATA_LCEVC` not covered
┃        >   --> /build/wl-screenrec-0.1.4-unstable-2024-07-28-vendor.tar.gz/ffmpeg-next/src/codec/packet/side_data.rs:70:15
┃        >    |
┃        > 70 |         match value {
┃        >    |               ^^^^^ patterns `sys::AVPacketSideDataType::AV_PKT_DATA_FRAME_CROPPING` and `sys::AVPacketSideDataType::AV_PKT_DATA_LCEVC` not covered
┃        >    |
┃        > note: `sys::AVPacketSideDataType` defined here
┃        >   --> /build/source/target/x86_64-unknown-linux-gnu/release/build/ffmpeg-sys-next-4f9a9cbd34c5c7ed/out/bindings.rs:3:275528
┃        >    |
┃        > 3  | ...)] pub enum AVPacketSideDataType { # [doc = " An AV_PKT_DATA_PALETTE side data packet contains exactly AVPALETTE_SIZE\n bytes worth of palette. This side data signals th…
┃        >    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                         …
┃        >    = note: the matched value is of type `sys::AVPacketSideDataType`
┃        > help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern, a match arm with multiple or-patterns as shown, or multiple match arms
┃        >    |
┃        > 126~             AV_PKT_DATA_AMBIENT_VIEWING_ENVIRONMENT => Type::AMBIENT_VIEWING_ENVIRONMENT,
┃        > 127~             sys::AVPacketSideDataType::AV_PKT_DATA_FRAME_CROPPING | sys::AVPacketSideDataType::AV_PKT_DATA_LCEVC => todo!(),
┃        >    |
┃        >
┃        > For more information about this error, try `rustc --explain E0004`.
┃        > error: could not compile `ffmpeg-next` (lib) due to 5 previous errors
┃        > warning: build failed, waiting for other jobs to finish...
┃        For full logs, run 'nix-store -l /nix/store/rqda27j6ycsb5zca3vvs4c5s5c0yg800-wl-screenrec-0.1.4-unstable-2024-07-28.drv'.
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@colemickens 

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.58, NixOS, 24.11 (Vicuna), 24.11.20241030.2d2a9dd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/0z5aw6jjbjq9sqcd03gfa7zkk25nzwgc-source`
```
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/wayland/wl-screenrec/default.nix
+++ b/pkgs/tools/wayland/wl-screenrec/default.nix
@@ -3,7 +3,7 @@
 , fetchFromGitHub
 , pkg-config
 , libdrm
-, ffmpeg
+, ffmpeg_6
 , wayland
 }:
 
@@ -28,7 +28,7 @@ rustPlatform.buildRustPackage rec {
   buildInputs = [
     wayland
     libdrm
-    ffmpeg
+    ffmpeg_6
   ];
 
   doCheck = false; # tests use host compositor, etc

